docs/cookbook/evals/evaluating_web_search_agent.md

```diff
@@ -0,0 +1,347 @@
+asyncio.run(run())
+# is_context_relevant=True explanation="The LLM-generated query effectively bridges the gap between the search history and the user query. It incorporates elements from both the search history (Thailand, beaches, cuisine) and the user's intent to book flights. The query maintains the original intent while adding relevant context from the search history, creating a coherent and specific request that aligns with the user's apparent travel interests. It successfully handles the topic shift by integrating the flight booking aspect with the previously established context of Thai beaches and cuisine, resulting in a semantically coherent and contextually enhanced query."
+```
+
+Now let's update our `llm_query`:
+
+```python
+llm_query = "General steps for booking flights online"
+asyncio.run(run())
+# is_context_relevant=False explanation="The LLM-generated query 'General steps for booking flights online' is not context-relevant. It addresses the user's specific question about booking flights but fails to incorporate any context from the search history about Thai beaches and cuisine. The query represents a complete topic shift without attempting to maintain relevance to the previous searches, missing an opportunity to provide more tailored advice such as booking flights to Thailand specifically."
+```
+
+We can verify that the `llm_query` does not mention anything related to the `search_history`, and therefore is properly labeled as context irrelevant.
+
+However, it's important to note that not all user queries need to be context-relevant to previous searches. Users may intentionally shift topics or ask unrelated questions in succession, which is a natural part of chatbot interactions.
+
+## Implementing the test
+
+Now that we have our evaluation, we can write our test.
+
+```python
+import pytest
+
+messages = [
+    {"role": "user", "content": "I am a SWE looking for a LLM dev tool library"},
+    {
+        "role": "assistant",
+        "content": "",
+        "tool_calls": [
+            {
+                "type": "function",
+                "function": {
+                    "arguments": '{"queries":["best LLM development tools","top libraries for LLM development","LLM libraries for software engineers","LLM dev tools for machine learning","most popular libraries for LLM development"]}',
+                    "name": "_web_search",
+                },
+                "id": "call_1",
+            }
+        ],
+    },
+    {
+        "role": "tool",
+        "content": "..."
+    }
+]
+
+@pytest.mark.asyncio
+async def test_conversation():
+    web_assistant = WebAssistant(
+        search_history=[
+            "best LLM development tools",
+            "top libraries for LLM development",
+            "LLM libraries for software engineers",
+            "LLM dev tools for machine learning",
+            "most popular libraries for LLM development",
```

Comment:
seems like this is duplicated from the tool message above, might be worth refactoring this?